### DISCUSSION TITLE
NanoELS with Nema Closed Loop

### DISCUSSION THREAD

**Discussion by HW61** [kachurovskiy/nanoels#43] [Show and tell]
Hello to everyone,
hello Maxim,

thank you very much for your ideas, work, videos and information. 

My name is Hermann, I am an elder beginner and started machining last fall. I was interested in cutting threads, not realy needing, but I wanted to know how. I changed gears and thougt, there should be a better solution. So I started searching and in December I discovered your project. 

I did not wait long and ordered pcb version 1 and all the other parts. As you suggested, I chosed a nema 24 closed loop stepper motor, including the ct57l stepper driver by stepperonline (they offer very good customer service). Because of your description and provided .stl files I started, with no idea what arduino is and what it does, I ordered the parts and started. After starting to build I found our about pcb version 2. So pcb 2 was ordered. At the same time printing the .stl files (as far as needed) started.

To my luck, your lathe is very similar to my Holzmann ED400FD. I modified some of the parts (motor pulley got a 3 mm longer shaft, motor mount got a bottom part to take up the motor and the encoder gear, which I printed with 80 teeth. With a very little struggle to get the display work (I had to solder wires from nano to the display) the ELS was finished and works. Today I did the first tests. It seems to work propperly, the machinist has to get used to operate the ELS.

Coming to an end, I have to say thank you very much again. It is not only for the project itself, it is also for the fact, you made me to get in contact with CAD (FreeCad) and 3d printing. Thank you for the challange.

With regards to all

Herman
![IMG_20220204_134209](https://user-images.githubusercontent.com/97787161/153288846-adb9a108-42ee-49c2-9af9-016c9cdf29ad.jpg)
![IMG_20220126_173830](https://user-images.githubusercontent.com/97787161/153288857-6e06d1db-3130-45a9-a400-1b4778aee91c.jpg)
![IMG_20220204_114405](https://user-images.githubusercontent.com/97787161/153288859-202c4174-355e-4aab-9acc-a164fcdff795.jpg)
![IMG_20220204_114417](https://user-images.githubusercontent.com/97787161/153288873-dd051327-3b16-495d-a30e-c35fb1c93cee.jpg)
![IMG_20220204_114701](https://user-images.githubusercontent.com/97787161/153288884-44b59ed1-88fb-46a6-9886-7a21615d630a.jpg)
![IMG_20220204_115423](https://user-images.githubusercontent.com/97787161/153288897-ecd85025-0eb4-4452-bfac-02a6412b499f.jpg)
![IMG_20220204_122043](https://user-images.githubusercontent.com/97787161/153288905-3b9986ef-bf9a-408c-acfd-5bc84a61cf47.jpg)
![IMG_20220204_134029](https://user-images.githubusercontent.com/97787161/153288911-94346cc9-ec6f-4ecb-bcea-078d9df5071d.jpg)
n

**Comment by kachurovskiy** [comment]
Herman, what an awesome build! Especially the colorful 3D prints. Big thanks for sharing your project 🎉

**Reply by deanZZZZZ** [reply]
Can you share the files to print? I have the same lathe and I'm deciding to upgrade :)

**Reply by Nitrous2000** [reply]
Herman,  a bit off topic, but as an "elder" newbie to CAD, I'm inspired by your work!  
You give me hope!

**Reply by HW61** [reply]
Thanks, it is a pleasure for me

**Comment by HW61** [comment]
Hello deanZZZZZ,

 

of course I do. As far as I “constructed” parts myself, I do also send Freecad files, in case you need changes. This are the files I used. Files will be send in several mails.

 

Hope the files help.

 

Hermann

 

 

Von: deanZZZZZ ***@***.***> 
Gesendet: Montag, 13. März 2023 11:31
An: kachurovskiy/nanoels ***@***.***>
Cc: HW61 ***@***.***>; Author ***@***.***>
Betreff: Re: [kachurovskiy/nanoels] NanoELS with Nema Closed Loop (Discussion #43)

 

Can you share the files to print? I have the same lathe and I'm deciding to upgrade :)

**Reply by kachurovskiy** [reply]
Feel free to drag and drop the files into the comment box on https://github.com/kachurovskiy/nanoels/discussions/43, they will be attached and shared 👍 Thanks!

**Reply by HW61** [reply]
Hello Maxim,
I tried to upload .stl files, but they are not supported. What can I do?
Thans a lot

**Reply by kachurovskiy** [reply]
@HW61 try archiving them into a ZIP file and uploading it

**Reply by HW61** [reply]
Thanks. Worked perfect

**Comment by deanZZZZZ** [comment]
OK :)
thank you

**Reply by HW61** [reply]
Done

**Comment by deanZZZZZ** [comment]
hmmm... where can I download the files?

**Comment by HW61** [comment]
I thought they were directly mailed to you. I will upload them to discussion #43 tonight.Am 19.03.2023 12:42 schrieb deanZZZZZ ***@***.***>:
hmmm... where can I download the files?

**Reply by deanZZZZZ** [reply]
ok

**Comment by HW61** [comment]
[Neue_Nanoels_Dateien.zip](https://github.com/kachurovskiy/nanoels/files/11012031/Neue_Nanoels_Dateien.zip)

**Reply by deanZZZZZ** [reply]
thx

**Reply by deanZZZZZ** [reply]
why is Gear_Encoder_big so big (80T)? Isn't a 1:1 ratio necessary?

**Reply by HW61** [reply]
There is no 1:1 ratio necessary. I have chosen a 1:2 ratio. This has to be observed in the driver settings

**Reply by HW61** [reply]
I have chosen this ratio, to fit the encoder to the machine

**Reply by deanZZZZZ** [reply]
My lathe ED400FD (without DIG) has 40 teeth.
Yours the same?
Where in the code should I change the ratio?
I'm waiting for a 600 encoder.

**Reply by kachurovskiy** [reply]
@deanZZZZZ just count how many encoder pulses would there be for 1 spindle revolution and use that number in line 6 - https://github.com/kachurovskiy/nanoels/blob/main/h2/h2.ino#L6

E.g. spindle with 40 teeth rotating gear with 80 teeth on 600 ppr encoder would yield 40/80*600 = 300 pulses for 1 spindle turn. Cheers!

**Reply by deanZZZZZ** [reply]
ok

**Comment by HW61** [comment]
Yes, 40Am 20.03.2023 14:17 schrieb deanZZZZZ ***@***.***>:
My lathe ED400FD (without DIG) has 40 teeth.
Yours the same?
